packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -140,10 +146,19 @@ class DatePicker extends React.Component {
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.selectedDate !== prevState.prevPropsSelectedDate) {
+    const selectedDate = nextProps.value || nextProps.selectedDate;
```

Comment:
Is an empty string a valid value? If so this will fall through and create an uncontrolled component. This may behave strangely and switch between controlled and uncontrolled within the same workflow. If an empty string is valid we may need to update the logic check. If it's not valid this should be fine.

Reply:
Yes, an empty is a valid value used to clear the date. Updated the logic to accept an empty string as the value: https://github.com/cerner/terra-framework/pull/738/commits/8a0d1a460950bd9fb1406f89ca4bdd73932bd2f0